ludwig/api.py

```diff
@@ -1898,7 +1898,13 @@ def save_config(self, save_path: str) -> None:
         """
         os.makedirs(save_path, exist_ok=True)
         model_hyperparameters_path = os.path.join(save_path, MODEL_HYPERPARAMETERS_FILE_NAME)
-        save_json(model_hyperparameters_path, self.config_obj.to_dict())
+        save_json(model_hyperparameters_path, self._scrub_creds(self.config_obj).to_dict())
+
+    def _scrub_creds(self, config_obj: ModelConfig) -> ModelConfig:
+        """Returns a copy of the config with all sensitive fields scrubbed."""
+        config_dict = config_obj.to_dict()
+        config_dict.get("backend", {})["credentials"] = {}
+        return ModelConfig.from_dict(config_dict)
```

Comment:
@Infernaught i think what we want to do is to temporary store credentials in a variable, write the artifact to disk, and then insert the credentials back into the object. 

In most cases, we do need to persist these credentials in memory in the process since we will likely use it downstream, but we just want to make sure it doesn't make it to disk.

Reply:
Ah I see, nvm! Looks good to me